packages/cozy-intent/src/api/services/NativeService.spec.ts

```diff
@@ -0,0 +1,140 @@
+import { WebviewRef } from '../models/environments'
+import { NativeEvent } from '../models/events'
+import { NativeMethodsRegister } from '../models/methods'
+import { NativeMessenger } from '../services/NativeMessenger'
+import { NativeService } from './NativeService'
+
+jest.mock('../services/NativeMessenger')
+
+const onMessageMock = jest.fn()
+class MockNativeMessenger extends NativeMessenger {
+  constructor(webviewRef: WebviewRef) {
+    super(webviewRef)
+  }
+
+  public postMessage = jest.fn()
+
+  public addMessageListener = jest.fn()
+
+  public onMessage = (event: NativeEvent): void => {
+    onMessageMock(event)
+  }
+}
+
+describe('NativeMessenger', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('Should allow to register and unregister webviews', async () => {
+    const nativeMethods: NativeMethodsRegister = {
+      logout: jest.fn(),
+      openApp: jest.fn()
+    }
+
+    const webviewRef: WebviewRef = {
+      injectJavaScript: jest.fn(),
+      props: {
+        source: {
+          uri: 'http://SOME_URI'
+        }
+      }
+    }
+
+    const nativeService = new NativeService(nativeMethods, MockNativeMessenger)
+
+    nativeService.registerWebview(webviewRef)
+
+    await nativeService.tryEmit({
+      nativeEvent: { data: '{"source": "post-me"}', url: 'http://SOME_URI' }
+    })
+
+    expect(onMessageMock).toHaveBeenNthCalledWith(1, {
+      nativeEvent: { data: '{"source": "post-me"}', url: 'http://SOME_URI' }
+    })
+
+    nativeService.unregisterWebview(webviewRef)
+
+    await expect(
+      nativeService.tryEmit({
+        nativeEvent: { data: '{"source": "post-me"}', url: 'http://SOME_URI' }
+      })
+    ).rejects.toThrow()
```

Comment:
nit: it throws nothing?

Reply:
Fixed in https://github.com/cozy/cozy-libs/pull/1432/commits/996dc4b62a9699f6fd65315a5c28c2df43067ada